fakes_test.go

```diff
@@ -7,13 +7,51 @@
 package fakesuserstudy
 
 import (
+	"context"
+	"fmt"
 	"testing"
 
 	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/arm"
+	fakeazcore "github.com/Azure/azure-sdk-for-go/sdk/azcore/fake"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
 	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/compute/armcompute/v4"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/compute/armcompute/v4/fake"
 	"github.com/stretchr/testify/require"
 )
 
+// matthchr: I don't like that this has to be package-scoped and not scoped to the individual test. There's gonna be a proliferation of these in test files.
+type fakeGetServer struct {
+	fake.VirtualMachinesServer
+
+	getVMResponse armcompute.VirtualMachine
+}
+
+// matthchr: No autocompletion of method name or signature, which is annoying, because no way for me to signify that I want to override a particular method. I need to go
+// copy the method from fake.VirtualMachinesServer. Obviously I can do that it'd be nicer if I could intellisense it. Especially because when I copy it I also have to go fix up
+// the imports for Responder[...] and ErrorResponder to be fakeazcore.Responder and fakeazcore.ErrorResponder.
+// matthchr: Actually, I can't replace with fakeazcore.Responder and fakeazcore.ErrorResponder, I need to replace with fake.Responder and fakeazcore.Responder... why are there two copies?
```

Comment:
I received similar feedback from other folks as well.

The copy in the `armcompute/fake` package contains a subset of the APIs from the one in `azcore/fake`.  I did this to remove the APIs that are intended to be called from the generated code.  It's looking like this is a bit over-engineered and just causes confusion.  It also breaks the copy/paste pattern which I really want to work.